aggregator/shard.go

```diff
@@ -87,3 +147,11 @@ func (s *aggregatorShard) Close() {
 	s.closed = true
 	s.metricMap.Close()
 }
+
+func (s *aggregatorShard) checkWritable() error {
```

Comment:
nit: perhaps `checkWritableWithLock` would be more appropriate since access to `s.latestWriteableNanos` is guarded with a lock?

Reply:
Right yeah sounds good.